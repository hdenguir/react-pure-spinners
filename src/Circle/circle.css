:root {
  --rps-circle-color: #ccc;
  --rps-circle-size: 64px;
}

.rps-circle {
  display: inline-block;
  transform: translateZ(1px);
}
.rps-circle > div {
  display: inline-block;
  width: var(--rps-circle-size);
  height: var(--rps-circle-size);
  margin: 8px;
  border-radius: 50%;
  background: var(--rps-circle-color);
  animation: rps-circle 2.4s cubic-bezier(0, 0.2, 0.8, 1) infinite;
}
@keyframes rps-circle {
  0%,
  100% {
    animation-timing-function: cubic-bezier(0.5, 0, 1, 0.5);
  }
  0% {
    transform: rotateY(0deg);
  }
  50% {
    transform: rotateY(1800deg);
    animation-timing-function: cubic-bezier(0, 0.5, 0.5, 1);
  }
  100% {
    transform: rotateY(3600deg);
  }
}
