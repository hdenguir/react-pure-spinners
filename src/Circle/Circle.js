import './circle.css'
import React from 'react'
import PropTypes from 'prop-types'
import { clsx } from '../utils'

const Circle = ({ className, color, size, style }) => {
  const varStyle = { ...style }
  color && (varStyle['--rps-circle-color'] = color)
  size && (varStyle['--rps-circle-size'] = `${size}px`)

  return (
    <div className={clsx('rps-circle', className)} style={varStyle}>
      <div />
    </div>
  )
}

Circle.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.number,
  style: PropTypes.object
}

export default Circle
