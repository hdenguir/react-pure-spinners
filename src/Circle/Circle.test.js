import React from 'react'
import renderer from 'react-test-renderer'
import Circle from './Circle'

it('renders correctly', () => {
  const tree = renderer.create(<Circle />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('joins default class with className prop', () => {
  const { root } = renderer.create(<Circle className="spin" />)

  expect(root.findByType('div').props.className).toEqual('rps-circle spin')
})

it('sets CSS vars with props', () => {
  const { root } = renderer.create(
    <Circle
      size={100}
      color="#d36ac2"
      style={{ margin: 20 }}
      className="spin"
    />
  )
  const { style } = root.findByType('div').props

  expect(style).toEqual({
    '--rps-circle-size': '100px',
    '--rps-circle-color': '#d36ac2',
    margin: 20
  })
})
