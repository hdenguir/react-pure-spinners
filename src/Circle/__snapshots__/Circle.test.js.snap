// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rps-circle"
  style={Object {}}
>
  <div />
</div>
`;
